daemon/analytics.c

```diff
@@ -1,5 +1,41 @@
 #include "common.h"
 
+void analytics_main_cleanup(void *ptr) {
+    struct netdata_static_thread *static_thread = (struct netdata_static_thread *)ptr;
+    static_thread->enabled = NETDATA_MAIN_THREAD_EXITING;
+
+    debug(D_ANALYTICS, "Cleaning up...");
+
+    static_thread->enabled = NETDATA_MAIN_THREAD_EXITED;
+}
+
+void *analytics_main(void *ptr) {
+    netdata_thread_cleanup_push(analytics_main_cleanup, ptr);
+    int sec = 0;
+
+    debug(D_ANALYTICS, "Analytics thread starts");
+
+    while(!netdata_exit) {
+        sleep(1);
```

Comment:
A suggestion, please, use Netdata `heartbeat`  instead the sleep.

Reply:
Done! With https://github.com/netdata/netdata/pull/10887/commits/45c2923382fee1289c7a33f585f1973f895ba5c7